src/GovUk.Education.ExploreEducationStatistics.Content.Services/DataSetFileService.cs

```diff
@@ -246,6 +248,37 @@ public async Task<Either<ActionResult, DataSetFileViewModel>> GetDataSetFile(Gui
         };
     }
 
+    public async Task<ActionResult> DownloadDataSetFile(
+        Guid dataSetFileId)
+    {
+        var releaseFile = await _contentDbContext.ReleaseFiles
+            .Include(rf => rf.File)
+            .Where(rf =>
+                rf.File.DataSetFileId == dataSetFileId
+                && rf.ReleaseVersion.Published.HasValue
+                && DateTime.UtcNow >= rf.ReleaseVersion.Published.Value)
```

Comment:
Is there a situation where something would be "published" in the future? IIRC a scheduled publish uses a different column.

Reply:
Not sure to be honest. Without looking it up, my guess would be no - I asked on Slack and one other person agreed. So you could argue this is redundant, but I guess since we absolutely don't want to be displaying releases that shouldn't be published, there is no harm in this.